### PR TITLE
chore(graphene): validate graphene support

### PR DESCRIPTION
**Pull request by mabdinur**
## Description 

Adds snapshot tests to ensure the graphql integration is compatible with for graphene v2 and v3

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Review by P403n1x87 — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -738,6 +738,13 @@ jobs:
       - run_tox_scenario:
           pattern: '^gevent_contrib-'
 
+  graphene:
+    <<: *contrib_job_small
```
Comment (mabdinur) [inline .circleci/config.yml:742]:
```suggestion
    <<: *machine_executor
```

File: tests/snapshots/tests.contrib.graphene.test_graphene.test_schema_failing_execute_v2.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+       "name": "graphql.execute",
+       "service": "graphql",
+       "resource": "query something{ patron { id name age } }",
+       "trace_id": 0,
+       "span_id": 4,
+       "parent_id": 1,
+       "type": "graphql",
+       "error": 1,
+       "meta": {
+         "error.msg": "exception was raised in a graphene query",
```
Comment (brettlangdon) [inline tests/snapshots/tests.contrib.graphene.test_graphene.test_schema_failing_execute_v2.json:60]: are these suppose to be so bare/useless?

File: tests/contrib/graphene/test_graphene.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+@pytest.mark.asyncio
+@pytest.mark.skipif(graphene.VERSION < (3, 0, 0), reason="execute_async is only supported in graphene>=3.0")
+async def test_schema_execute_async(test_schema, test_source_str):
+    result = await test_schema.execute_async(test_source_str)
+    assert not result.errors
+    assert result.data == {"patron": {"id": "1", "name": "Syrus", "age": 27}}
+
+
+@pytest.mark.snapshot
+def test_schema_execute(test_schema, test_source_str, enable_graphql_resolvers):
```
Comment (brettlangdon) [inline tests/contrib/graphene/test_graphene.py:75]: should we have variants for with and without resolvers?

async vs not-async traces should always be identical (other than whether resolvers are enabled or not), correct?

**Review by brettlangdon — COMMENTED** [review]
some minor comments/thoughts, but otherwise nice! I like it.

File: tests/snapshots/tests.contrib.graphene.test_graphene.test_schema_failing_execute_v2.json (hunk 2 above)
Comment (mabdinur) [reply tests/snapshots/tests.contrib.graphene.test_graphene.test_schema_failing_execute_v2.json:60]: Do you mean the error message? I am setting that error message here: https://github.com/DataDog/dd-trace-py/blob/90ca0bcc7e9a9bd9f25efe33b7fd8101042fcf0f/tests/contrib/graphene/test_graphene.py#L26

File: tests/contrib/graphene/test_graphene.py (hunk 3 above)
Comment (mabdinur) [reply tests/contrib/graphene/test_graphene.py]: Yup, I just made this change

File: tests/contrib/graphene/test_graphene.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+@pytest.mark.snapshot(token="tests.contrib.graphene.test_graphene.test_schema_execute")
+@pytest.mark.skipif(graphene.VERSION < (3, 0, 0), reason="execute_async is only supported in graphene>=3.0")
+async def test_schema_execute_async(test_schema, test_source_str):
+    result = await test_schema.execute_async(test_source_str)
+    assert not result.errors
+    assert result.data == {"patron": {"id": "1", "name": "Syrus", "age": 27}}
+
+
+@pytest.mark.snapshot
+def test_schema_execute_with_resolvers(test_schema, test_source_str, enable_graphql_resolvers):
```
Comment (brettlangdon) [inline tests/contrib/graphene/test_graphene.py:82]: should we do the same with async?

**Review by brettlangdon — APPROVED** [review]

**Review by Kyle-Verhoog — COMMENTED** [review]
Can we add this to our list of integrations + a release note

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**